src/UI/Actions.hs

```diff
@@ -1381,17 +1387,12 @@ buildMail s = do
 -- the configuration as another? type variable. So all in all it was non
 -- trivial.
 -- See #200
-trySendAndCatch :: String -> B.ByteString -> AppState -> IO AppState
-trySendAndCatch l' m s = do
-    let cmd = view (asConfig . confComposeView . cvSendMailCmd) s
-        defMailboxes = view (asConfig . confComposeView . cvIdentities) s
-    catch
-        (cmd m $> (s
-         & set asCompose (initialCompose defMailboxes)
-         . set (asConfig . confBoundary) l'))
-        (\e ->
-              let err = show (e :: IOException)
-              in pure $ s & setError (SendMailError err))
+trySendAndCatch :: (MonadState AppState m, MonadIO m, MonadCatch m) => B.Builder -> m ()
```

Comment:
I have the feeling that the above comment can be deleted with the refactoring? Seems like this change is solving the problem I had back then?